test/export/test_export.py

```diff
@@ -420,6 +420,28 @@ def forward(self, x):
         ):
             ep.module()(torch.tensor([3]))
 
+    def test_container_leak(self):
+        class Bar(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self._cache = {}
+
+            def forward(self, x):
+                self._cache["leaky"] = x.sum()
+                return x.sum()
+
+        class Foo(torch.nn.Module):
+            def __init__(self, bar):
+                super().__init__()
+                self.bar = bar
+
+            def forward(self, x):
+                return self.bar(x)
+
+        foo = Foo(Bar())
+        with self.assertRaisesRegex(ValueError, "self.bar._cache"):
```

Comment:
@avikchaudhuri I am bit unsure what the correct behaviour here should be. Should we just warn to be consistent with other checks I have in this PR? If we decide to just warn whenever something sketchy happens, we could relax https://github.com/pytorch/pytorch/blob/8b78ba07b14d774f889f3b863a766e65d60cabff/torch/_functorch/_aot_autograd/frontend_utils.py#L280 these ones as well. Wdyt?

Reply:
It will just capture the tensor computations so bar._cache and operations on them won't show up. So we are kind of dropping side effects.